src/containers/forms/PractitionerForm/index.tsx

```diff
@@ -0,0 +1,246 @@
+import { ErrorMessage, Field, Form, Formik } from 'formik';
+import React, { useState } from 'react';
+import { Redirect } from 'react-router';
+import UserIdSelect, { Option } from './UserIdSelect';
+
+import moment from 'moment';
+import { Button, Label } from 'reactstrap';
+import { FormGroup } from 'reactstrap';
+import * as Yup from 'yup';
+import {
+  ACTIVE,
+  NAME,
+  NAME_REGEX_ERROR,
+  NO,
+  OPENSRP_PRACTITIONER_ENDPOINT,
+  PRACTITIONER,
+  PRACTITIONERS_LIST_URL,
+  REQUIRED,
+  SAVE,
+  SAVING,
+  USERNAME,
+  YES,
+} from '../../../constants';
+import { generateNameSpacedUUID } from '../../../helpers/utils';
+import { OpenSRPService } from '../../../services/opensrp';
+
+/** Namespace for generating uuids */
+const PractitionerFormNameSpace = '78295ac0-df73-11e9-a54b-dbf5e5b2d668';
+
+export interface PractitionerFormFields {
+  identifier: string;
+  active: boolean;
+  name: string | undefined;
+  userId: string | undefined;
+  username: string | undefined;
+}
+
+/** Initial values for practitioner's form */
+export const defaultInitialValues: PractitionerFormFields = {
+  active: false,
+  identifier: '',
+  name: undefined,
+  userId: undefined,
+  username: undefined,
+};
+
+/** yup validations for practitioner data object from form */
+export const PractitionerSchema = Yup.object().shape({
+  active: Yup.boolean(),
+  identifier: Yup.string(),
+  name: Yup.string()
+    .matches(/^[a-zA-Z0-9]+([_ -]?[a-zA-Z0-9])*$/, NAME_REGEX_ERROR)
```

Comment:
Will this come back to bite us?  Why do we need to validate the name using a regular expression?

Reply:
will fix this